paraatm/io/weather_cube_ET.py

```diff
@@ -0,0 +1,366 @@
+#!/home/anaconda3 python
+# -*- coding: utf-8 -*-
+"""
+NASA NextGen NAS ULI Information Fusion
+
+@organization: Arizona State University
+@author: Yutian Pang
+@date: 2019-02-19
+@last updated: 2020-06-12
+
+This script is used to read and process the EchoTop (ET) convective weather products from
+the Corridor Integrated Weather System (CIWS). The data can be downloaded from NASA
+Sherlock Data Warehouse. url: https://ntrs.nasa.gov/archive/nasa/casi.ntrs.nasa.gov/20180004248.pdf
+
+The script takes raw ET convective weather files and the trajectory file as inputs. The outputs are
+the processed weather feature cube around each of the coordinates in trajectory file, as well as the
+value of the ET right at each coordinate.
+
+Minimal changes required to process VIL convective weather feature in a similar way.
+"""
+
+import os
+import time
+import math
+import datetime
+import numpy as np
+import pandas as pd
+from netCDF4 import Dataset
+
+
+class WeatherCubeGenerator(object):
+    """
+        Read CIWS EchoTop file and return weather values at and around the coordinates based on the trajectory file
+        The input to this class should be a dictionary.
+
+        The header of the trajectory file should contain:
+
+        1. UNIX TIME
+        2. LATITUDE
+        3. LONGITUDE
+
+        The raw EchoTop file should be multiple .nc files updated every 150 seconds
+
+        Sample data can be found under './sample_data/weather_parser' directory
+
+        Parameters
+        ----------
+        cube_size : int
+            size of the weather tensor to take around the current location. If set 20, the return of weather_area
+            will have dimension of len(trajectory)-1 x 20 x 20 x 1. And the return of weather_point will have
+            dimension of len(trajectory)-1 x [index_x, index_y, value].
+        resize_ratio : int
+            The ratio to resize the raw ET weather grid. The original resolution is 3520 x 5120. Increase this
+            value will make the grid coarser, but increase the area covered around the current location. If set
+            10, the raw ET weather map will first resize to resolution 352 x 512, then take a 20 x 20 grid area.
+        downsample_ratio : int
+            The ratio to equally downsample the trajectory file to reduce the length of the trajectory and reduce
+            time to process the data. If the original length of the whole trajectory is 500, set downsample_ratio=10
+            means the script will first equally sample 50 points in the whole trajectory ,then process the weather
+            data based on the sampled-trajectory.
+        weather_path: str
+            The path to the weather data.
+        trajectory_path: str
+            The path to the trajectory data.

```

Comment:
This is excellent documentation and very helpful.  Two suggestions:

1. Although it is done both ways, I suggest moving description of the parameters to the `__init__` function.  This is consistent with [PEP 257](https://www.python.org/dev/peps/pep-0257/) that "The class constructor should be documented in the docstring for its `__init__` method."  I also find this easier when reading the code, since documentation of the arguments appears in the same place where they are defined in the code (the `__init__` function).  And it is consistent with documentation of other classes in para-atm.
2. As it currently stands, `__init__` actually accepts a dictionary.  However, if you make the other change that I suggested to expand out these arguments instead of passing a dictionary, then your documentation would match the arguments of `__init__`.